packages/loader/container-loader/src/container.ts

```diff
@@ -544,7 +570,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 },
                 error);
             });
-        this._audience = new Audience();
+        this._audience = audience ?? new Audience();
```

Comment:
If we go with this route, I'd suggest to creating Audience same way as Quorum is created - i.e. when protocol is initialized.
I think the reason it's done the way it's done - handling of "connect" event and operating on audience long before protocol is setup. I'd suggest changing it by synthetizing fake signals and pushing them into  this._deltaManager.inboundSignal. That way all processing and all initialization is symmetrical.

Reason I like it (independently from this PR) - people get really confused about audience and quorum. I think we own them revisiting this area, and ideally (if it's possible) - expose single data structure to track users in document. At least at Container (host interaction level), where we have a bit more room to "lie" by exposing user (clientId) presence in simpler way. Story for DDSs is different - there we rely on specific order of quorum operations (all consensus data structures), so we have to expose conceptually data roughly in same way. But that's unknown (maybe thing never to happen), so maybe we do not care?

Reply:
I'm wary of creating and sending signals in that part of the code. Also, the interface provided by the delta manager for it's signal queue is limited to reading so I'd have to open up the delta manager to allow for this.. I have opted for storing the existing clients explicitly and providing them to the builder, and at the same time add them to the audience iff the protocol handler is already instantiated. I believe it will have the same effect.